File: kdcreader_ios/ios/Classes/delegate/KDelegate.swift
//
//  KDelegate.swift
//  kdcreader_ios
//
//  Created by koamtac on 3/19/24.
//

import Flutter

protocol KDelegate: AnyObject {
    func isSupported(_ name: String) -> Bool
    func run(_ call: FlutterMethodCall, result: @escaping FlutterResult) -> Bool
}

File: kdcreader_android/android/src/main/kotlin/koamtac/kdc/sdk/plugins/delegate/KDelegate.kt
package koamtac.kdc.sdk.plugins.delegate

import android.content.Context
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

interface KDelegate {
    fun isSupported(action: String): Boolean
    fun run(context: Context, call: MethodCall, result: MethodChannel.Result): Boolean
}